### DISCUSSION TITLE
How do I pass data to the Header Create Action in an InfoList, based on the currently active tab?

### DISCUSSION THREAD

**Discussion by AntonioPrimera** [filamentphp/filament#13261] [Help] [answered]
### Package

Infolist builder

### Package Version

v3.0.0

### How can we help you?

The idea is pretty simple: I am building a website for a restaurant. In their menu, they have several food categories and each category has several items.

In the InfoList for the Menu Items, i created a dynamic Tab bar, with a tab for each category, so, for example, when the user clicks the "Soups" tab, only items belonging to the soups category are shown in the table below.

As expected, in the header, there is a CreateAction button which opens the form for creating a new item. The form contains a dropdown, where you select the category.

So, what I want to do: when the user clicks a tab in the info-list and then clicks the Create button, I would like to pass the selected tab (the name, or a category_id) to the form, so the category is pre-selected. Is there any way to do this, without getting very hacky (like writing to the local storage or to the session)?

**Comment by leandrocfe** [comment]
You could add the active tab as a query param

https://livewire.laravel.com/docs/url#nullable-properties

**Reply by AntonioPrimera** [reply]
In my Infolist I have the following (standard) way of adding header actions, and commented out the 2 lines of code which get the MenuCategory corresponding to the currently selected tab, but I couldn't figure out how to pass this to the action. I tried `extraAttributes(...)`, I tried adding the form schema with some parameters, but the information doesn't seem to get to the form.

```php
    protected function getHeaderActions(): array
    {
        //$currentTab = $this->activeTab;
        //$activeCategory = MenuCategory::where('name', $currentTab)->first();

        return [
            Actions\CreateAction::make(),
        ];
    }
```

While my Laravel and Livewire skills are pretty good, Filament is new for me and this is the first project (a fairly simple one), where I try it out and see what it can do and what its limits are.

**Reply by leandrocfe** [reply]
Where would you like to use `$activeCategory`? I mean, what value are you trying to change in the action?

**Reply by AntonioPrimera** [reply]
Currently I have the following class:

```php
class ListMenuItems extends ListRecords
{
    protected static string $resource = MenuItemResource::class;

    public function getTabs(): array
    {
        $tabs = [
            'all' => Tab::make('Toate'),
        ];

        foreach (MenuCategory::active()->get() as $category) {
            $tabs[$category->name] = Tab::make($category->name)
                ->modifyQueryUsing(function ($query) use ($category) {
                    $query->where('menu_category_id', $category->id);
                });
        }

        return $tabs;
    }

    protected function getHeaderActions(): array
    {
        //$currentTab = $this->activeTab;
        //$activeCategory = MenuCategory::where('name', $currentTab)->first();

        return [
            Actions\CreateAction::make(),
        ];
    }
}
```

And the CreateMenuItem Page, where I would guess that the parameter should be available:
```php

class CreateMenuItem extends CreateRecord
{
    protected static string $resource = MenuItemResource::class;
}
```

Or, ideally it would be available during setting up the form schema in the MenuItemResource, so that it's available when opening a new Page for creating an Item and also when the form is opened in a SlideOver:

```php
class MenuItemResource extends Resource
{
//...
    public static function form(Form $form): Form
    {
        //... I guess, it would be best to have it here
    }
}
```

**Reply by leandrocfe** [reply, marked as answer]
```php
->url(MenuItemResource::getUrl('create', ['activeTab' => $this->activeTab]))
```
`activeTab` query param will be available in your create page
